### DISCUSSION TITLE
Is it intentional that TCA is transitively dependent on OrderedCollections?

### DISCUSSION THREAD

**Discussion by kalupas226** [pointfreeco/swift-composable-architecture#1818] [Q&A] [answered]
Hi.

TCA imports `OrderedCollections` in [ForEachStore.swift](https://github.com/pointfreeco/swift-composable-architecture/blob/09271d56f48bfdaa6c7c598b73eedc5b8f49a75d/Sources/ComposableArchitecture/SwiftUI/ForEachStore.swift#L1).

However, TCA is only transitively dependent on OrderedCollections through IdentifiedCollections. 
https://github.com/pointfreeco/swift-composable-architecture/blob/09271d56f48bfdaa6c7c598b73eedc5b8f49a75d/Package.swift#L38

Is it intentional that TCA hasn't direct dependencies on OrderedCollections (swift-collections) ?

**Comment by mbrandonw** [comment, marked as answer]
I suppose it is intentional because we're not aware of a downside to depending on it transitively. Do you know of any?

**Reply by kalupas226** [reply]
Thank you for your quick response!

Here are downsides I can think of:
- You're depending on an other package's dependency so if that other package change its dependencies your code will break. Changing what a package depends on is not a breaking change (except when using types of a dependency in your public interface).
- From my understanding of [apple/swift-package-manager#5297](https://github.com/apple/swift-package-manager/issues/5297) it seems that being able to import a transitive dependency is a side effect of how SwiftPM currently works internally, not something that is explicitly allowed.

**Reply by stephencelis** [reply]
@kalupas226 Interesting, thanks for the clarification!

I don't think we're opposed to making the dependency explicit, it's just worked fine for now and hasn't been a priority to address (the SPM discussion is from 7 years ago, so I don't expect it to be addressed anytime soon 😅).

We'd happily take a pull request, though, if you'd like to submit one!

**Reply by davdroman** [reply]
> the SPM discussion is from 7 years ago, so I don't expect it to be addressed anytime soon 😅

For context, I think it might be in the cards for Swift 6 to actually address it https://forums.swift.org/t/design-priorities-for-the-swift-6-language-mode/62408/1:

> For example, Swift's module system might be improved to limit the effect of a module's implementation dependencies on its client modules, making modules and packages more resilient to change.

Anyway, Swift 6 is not expected to come in 2023 at all, so nothing to worry about just yet.

**Reply by kalupas226** [reply]
@stephencelis 

I see, thanks for the clarification!

> We'd happily take a pull request, though, if you'd like to submit one!

I'm happy to hear that 😄
I'd like to submit a pull request in a little bit.

@davdroman 

> For context, I think it might be in the cards for Swift 6 to actually address it https://forums.swift.org/t/design-priorities-for-the-swift-6-language-mode/62408/1:

I didn't know this information, thank you very much!

> Anyway, Swift 6 is not expected to come in 2023 at all, so nothing to worry about just yet.

Now I also understand that there is nothing to worry about just yet.
Thank you again.
